doc/server_configurations.md

```diff
@@ -7695,14 +7695,50 @@ Volar can be installed via npm
 npm install -g @volar/server
 ```
 
-With Vue 3 projects - it works out of the box.
+Vue 3 works by default, but Vue 2 needs [additional configuration](https://github.com/johnsoncodehk/volar/blob/master/extensions/vscode-vue-language-features/README.md?plain=1#L28-L63)
 
-With Vue 2 projects - requires [additional configuration](https://github.com/johnsoncodehk/volar#using)
+`vuels` and `volar` shouldn't be running at the same time
 
-Do not run `vuels` and `volar` at the same time.
+**[Take Over Mode](https://github.com/johnsoncodehk/volar/discussions/471)**
 
-To check which language servers are running, open a `.vue` file and run the `:LspInfo` command.
+To enable:
 
+```lua
+require'lspconfig'.volar.setup{filetypes = {'typescript', 'javascript', 'javascriptreact', 'typescriptreact', 'vue', 'json'}}
+```
+
+Remember to **disable** `tsserver` — Volar's Take Over Mode replaces it.
+
+**Custom TypeScript detection**
+The default config looks for TS in local node_modules. The alternatives to this behavior are:
+
+* use global TS - `require'lspconfig'.volar.setup{init_options = { typescript = { serverPath = '/home/[yourusernamehere]/.npm/lib/node_modules/typescript/lib/tsserverlib.js' } }}`
+* use global TS only if local TS not found
+```lua
+local util = require 'lspconfig/util'
+
+local function get_typescript_server_path(root_dir)
+  local project_root = util.find_node_modules_ancestor(root_dir)
+
+  local local_tsserverlib = project_root ~= nil and util.path.join(project_root, 'node_modules', 'typescript', 'lib', 'tsserverlibrary.js')
+  local global_tsserverlib = '/home/[yourusernamehere]/.npm/lib/node_modules/typescript/lib/tsserverlibrary.js'
+
+  if local_tsserverlib and util.path.exists(local_tsserverlib) then
+    return local_tsserverlib
+  else
+    return global_tsserverlib
+  end
+end
+
+require'lspconfig'.volar.setup{
+  config = {
+    on_new_config = function(new_config, new_root_dir)
+      new_config.init_options.typescript.serverPath = get_typescript_server_path(new_root_dir)
```

Comment:
Is there any advantage of automating this other than just having users fill this in directly? Adds a lot of code/ complexity for an example. I feel like hardcoding the string is easier.

Reply:
The global setting of TS helps when you open a .vue file that's not part of any project. If you created a .vue file in your home directory just to quickly draft something the default config will make the volar's binary throw an error and close because it didn't find TypeScript - it's unlikely there's node_modules with TypeScript in your home dir. Setting it globally fixes that problem.

The "use global if local TS not found" is useful when your global TS version is eg. super new, like TypeScript 4.5, but the one in your workspace is older - e.g. TypeScript 3.7.

Reply:
Right, I was suggesting shifting the responsibility of setting this correctly per project onto the user rather than embedding the utility function :)

Reply:
Users could theoretically set per-project init.lua files in their projects but I believe this is brittle and something only the non-mortal users of nvim do. 

I assume you don't want the CONFIG.md to baloon in size due to every LS providing their own user-experience improving tricks? I could:
1. just change the second bullet point (the one with the big code block) to something like "merge the existing TS detection function and the above global TS example to use global TS only when local TS is not found" and remove the codeblock
2. move the codeblock to a gist and link it in the docs